### PR TITLE
feat(gms): Adding yaml configuration for metadata-service

### PR DESCRIPTION
**Pull request by jjoyce0510**
This PR introduces a single YAML file (application.yml) where we can place all gms application configs. This will make understanding Metadata Service configuration much easier for folks! Going forward, any new features should derive their config from this YAML file. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java [hunk 1]
```diff
@@ -58,13 +62,17 @@
...
-    RestClientBuilder restClientBuilder;
 
+    System.out.println(
+        String.format("Hey we created it! %s %s %s %s %s", this.host, this.port, this.useSSL, this.username, this.password));
+
+
+    RestClientBuilder restClientBuilder;
     if (useSSL) {
-      restClientBuilder = loadRestHttpsClient(host, port, pathPrefix, threadCount, connectionRequestTimeout, sslContext, username,
+      restClientBuilder = loadRestHttpsClient(host, port, pathPrefix, 1, connectionRequestTimeout, sslContext, username,
```
Comment (dexter-mh-lee) [inline metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java:72]: why are we hardcoding to 1?

**Review by dexter-mh-lee — APPROVED** [review]
This is amazing! LGTM except for one piece

File: metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java (hunk 1 above)
Comment (jjoyce0510) [reply metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java:72]: Oops - thanks for catching this - remnant from some debugging !

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**